src/core/config/openthread-core-config-check.h

```diff
@@ -480,6 +480,21 @@
 
 #endif // OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
 
+#if OPENTHREAD_CONFIG_CSL_TRANSMITTER_ENABLE
+#if (OPENTHREAD_CONFIG_THREAD_VERSION < OT_THREAD_VERSION_1_2)
+#error "At least Thread Version 1.2 is required for OPENTHREAD_CONFIG_CSL_TRANSMITTER_ENABLE"
+#endif
+#if (!OPENTHREAD_FTD)
+#error "FTD is required for OPENTHREAD_CONFIG_CSL_TRANSMITTER_ENABLE"
+#endif
+#endif // OPENTHREAD_CONFIG_CSL_TRANSMITTER_ENABLE
```

Comment:
I think we can keep it here and maybe later we can make it configurable for users and it may detect some errors if we change our build for 1.2 incorrectly.